functorch/functorch/csrc/BatchRulesHelper.h

```diff
@@ -469,4 +469,9 @@ inline VmapDimVector range(int64_t start, int64_t stop) {
   return dims;
 }
 
+std::tuple<Tensor, Tensor> _binary_pointwise_helper(
+    const Tensor& tensor, optional<int64_t> tensor_batch_dim, const Tensor& other, optional<int64_t> other_batch_dim);
+std::tuple<Tensor, Tensor> _binary_pointwise_helper(
+    const Tensor& tensor, optional<int64_t> tensor_batch_dim, const Tensor& other, optional<int64_t> other_batch_dim,
+    bool do_type_promotion);
```

Comment:
nit: `do_type_promotion` could just be a default argument in C++, right? Though I know that some people prefer not to use default arguments, so totally up to you